compiler/ballerina-parser/src/test/resources/types/array-type/array_type_source_17.bal

```diff
@@ -5,4 +5,13 @@ public function foo() {
     int[]] a;
     int[&%] a;
     T 5] a;
+    int[1*2] intArr = [1];
+    T[*] = 6;
+    int[2] = [1,2];
```

Comment:
Shall we add a valid syntax test for array-type-desc with identifier? e.g. `arr[3]` .  Looks like we don't have one.

Reply:
Will add